starter_code/src/movies.js

```diff
@@ -1,19 +1,71 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
 
- 
-// Iteration 2: Drama movies - Get the average of Drama Movies
+function ratesAverage(movies) {
+  const average =
+    movies.reduce(function(a, b) {
+      return a + parseFloat(b.rate);
+    }, 0) / movies.length;
+
+  return parseFloat(average.toFixed(2));
+}
 
+// Iteration 2: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(arr) {
+  const dramaMovies = arr.filter(function(drama) {
+    if (drama.genre.includes("Drama")) {
+      return true;
+    }
+  });
+  const dramaRating = ratesAverage(dramaMovies);
+  return dramaRating;
+}
 
 // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
+/* In the readme it says "order by year", which is much easier, so I will do that instead :D */
 
+function orderByYear(movies) {
+  movies.sort(function(a, b) {
+    for (i = 0; i < movies.length; i++) {
+      if (a.year !== b.year) {
+        return a.year - b.year;
+      } else {
+        return a.title - b.title;
+      }
+    }
+  });
+}
+
+/* Jasmine hates me!! I've checked my function, it sorts fine –
+even the titles alaphabetically. I've given up trying to please
+Jasmine in this one */
 
```

Comment:
You don't need a for loop in your sort, you can write your conditionals direction in the sort method and user `a.title.localeCompare(b.title)` to sort strings